sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -71,6 +71,57 @@ def _succeeded(status):
     return str(status).lower() in _SUCCEEDED
 
 
+class OperationState(ABC):
+    @abc.abstractproperty
+    def finished(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def failed(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def succeeded(self):
+        # type: () -> list(str)
+        raise NotImplementedError()
+
+    def has_finished(self, status):
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in self.finished
+
+    def has_failed(self, status):
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in self.failed
+
+    def has_succeeded(self, status):
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in self.succeeded
+
+
+class DefaultOperationState(OperationState):
+    def __init__(self):
```

Comment:
You don't really need an init, you can just havet he propertyreturn the list. I.e.

```
@property
def finished(self):
    return frozenset(["succeeded", "canceled", "failed"])
```

Reply:
Done!